### PR TITLE
Lazy imports

### PR DESCRIPTION
**Pull request by arthurevans**
Iterating on @stramel's lazy-imports doc. 

cc: @rictic @katejeffreys

### CODE REVIEW

File: app/2.0/docs/glossary.md
Comment (abdonrd) [inline app/2.0/docs/glossary.md]: Then we should review and merge https://github.com/PolymerElements/polymer-starter-kit/pull/998.

File: app/2.0/docs/glossary.md
Comment (katejeffreys) [inline app/2.0/docs/glossary.md]: I suggest 

`### lazy imports`

instead of 

`### lazy-imports`

File: app/2.0/docs/glossary.md
Comment (katejeffreys) [inline app/2.0/docs/glossary.md]: This paragraph sounds a bit tautological to me :) I suggest rewording. Maybe something like this, edited for technical accuracy ;) :

"Lazy imports are a declarative way to describe the resources to be loaded lazily in an application (see #lazy-loading). Describing lazily-loaded resources declaratively means they can be linted, and that sharded bundling can be performed without configuration."

What is sharded bundling? We don't mention it in the glossary and idk what it means. I feel that we should define it in the glossary if we use it in another definition. 

Also not sure what it means to be able to do linting and sharded bundling without configuration. What config is needed if the resources are described imperatively? I am confuse :/

Not sure if others would share my confusion, however :)

File: app/2.0/start/toolbox/create-a-page.md
Comment (katejeffreys) [inline app/2.0/start/toolbox/create-a-page.md]: `on-demand` --> `on demand`

File: app/2.0/toolbox/lazy-imports.md
Comment (katejeffreys) [inline app/2.0/toolbox/lazy-imports.md]: I suggest deciding on "lazy-loaded" or "lazily-loaded" and sticking to it throughout :)

File: app/2.0/toolbox/lazy-imports.md
Comment (katejeffreys) [inline app/2.0/toolbox/lazy-imports.md]: "latest version of Polymer starter kit" --> "latest version of the Polymer starter kit"

File: app/2.0/toolbox/lazy-imports.md
Comment (katejeffreys) [inline app/2.0/toolbox/lazy-imports.md]: This is in the 2.0 section of the docs, so IMO we should remove the 1.0 syntax.

File: app/2.0/toolbox/prpl.md
Comment (katejeffreys) [inline app/2.0/toolbox/prpl.md]: "The app should lazy load code at runtime" --> "The app should lazy-load code at runtime as required to get a large performance" or maybe "The app should lazily load code at runtime as required to get a large performance"

File: app/2.0/toolbox/prpl.md
Comment (katejeffreys) [inline app/2.0/toolbox/prpl.md]: "ie." --> "e.g."

File: app/2.0/toolbox/prpl.md
Comment (katejeffreys) [inline app/2.0/toolbox/prpl.md]: "ie." --> "e.g."

**Review by ghost — APPROVED** [review]

File: app/2.0/docs/glossary.md
Comment (stramel) [inline app/2.0/docs/glossary.md]: Whoops, didn't notice this was on @arthurevans's branch :sweat_smile:

Comment (arthurevans) [inline app/2.0/docs/glossary.md]: @abdonrd Yeah, I was trying to get this ready so that we could roll it out whenever that got merged into the PSK.  (But we could also remove the sentence that says that the latest version of the PSK uses lazy imports. Let me know.)

Comment (abdonrd) [inline app/2.0/docs/glossary.md]: It would be great to have it merged into the PSK. 🙂

**Comment by katejeffreys** [comment]
@stramel Can you pls comment with a statement that you're OK with your commits being included in this PR?

I can go ahead and make the changes I suggested if nobody objects. 

Then this should be mergeable AFAIK. @arthurevans ?

**Comment by stramel** [comment]
@katejeffreys  I am okay with my commits being utilized in this or any PR :smile: 

The only concern I have with merging is that the polymer-starter-kit doesn't actually implement them yet, pending https://github.com/PolymerElements/polymer-starter-kit/pull/998.

There are a lot of things outstanding in the lazy-imports side that need review as well but nothing that should hold up this PR.

**Comment by abdonrd** [comment]
@arthurevans old PR; closing?

**Comment by stramel** [comment]
@arthurevans 	Should we go ahead and close this PR?

**Comment by arthurevans** [comment]
@stramel Yep, I think we should. Thanks for all your work on this.

**Pull request closed without merging**
